src/python/pants/backend/python/tasks/setup_py.py

```diff
@@ -42,12 +50,65 @@
 
 from setuptools import setup
 
-setup(**
-{setup_dict}
-)
+setup(**{setup_dict})
 """
 
 
+# Distutils does not support unicode strings in setup.py, so we must explicitly convert to binary
+# strings as pants uses unicode_literals. A natural and prior technique was to use `pprint.pformat`,
+# but that embeds u's in the string itself during conversion. For that reason we roll out own
+# literal pretty-printer here.
+#
+# For more information, see http://bugs.python.org/issue13943
+def distutils_repr(obj):
+  output = io.StringIO()
+  linesep = os.linesep
+
+  def _write(data):
+    output.write(ensure_text(data))
+
+  def _repr(o, indent=False, level=0):
```

Comment:
Perhaps rename this to better capture the side-effect and None return type, e.g. `_write_repr_formatted()`. I expected this to return a formatted `str`, and was confused below when I saw calls to `_repr()` without an assignment attached.